CMakeLists.txt

```diff
@@ -0,0 +1,45 @@
+cmake_minimum_required(VERSION 3.12)
+
+include(pico_sdk_import.cmake)
+include(pico_extras_import.cmake)
+
+project(dshot_test LANGUAGES C CXX ASM)
+
+set(CMAKE_C_STANDARD 11)
+set(CMAKE_CXX_STANDARD 17)
+
+add_compile_options(
+  -Wall
+  -Wno-format # int != int32_t as far as the compiler is concerned because gcc
+              # has int32_t as long int
+  -Wno-unused-function # we have some for the docs that aren't called
+  -Wno-maybe-uninitialized)
+
+pico_sdk_init()
+
+set(DSHOT_FILES ./lib/shoot/shoot.cpp ./lib/tts/tts.cpp ./lib/utils/utils.cpp
+                ./lib/dshot/dshot.cpp)
+
+add_executable(${PROJECT_NAME} src/main.cpp ${DSHOT_FILES}
+                               ./lib/extern/hx711-pico-c/src/hx711.c)
```

Comment:
Just a question (from someone who's a noob at CMake). Why do you add `hx711.c` in this line, instead of a library in `target_link_libraries`?

Reply:
I'm also a CMake noob :)

I think it's also possible to add all the c/c++ source files with `add_library` and `target_link_libraries` as detailed [here](https://cmake.org/cmake/help/v3.0/command/target_link_libraries.html), which would arguably be better as it may be a more modular way to share specific parts of the code as pre-compiled libraries. This is especially important for large projects where it may take a super long time to compile certain parts of the code.

But for our project, the code is small in size and to keep things simple, I just included them as source files in `add_executable`.